core/src/main/java/de/jplag/SubmissionFileData.java

```diff
@@ -3,11 +3,10 @@
 import java.io.File;
 
 /**
- * Contains the information about a single file in a submission. For single file submissions the submission file is the
- * same as the root.
```

Comment:
Note: I was unable to reproduce the behavior previously described here. It seems that the object `root` is just the same for all submissions, regardless of whether they are single-file submissions or submission directories.

Reply:
I think the comment means that both fields of type `File` are identical if it is a single file submission.

E.g. for `folder/file.java`:

- submissionFile = `file.java`
- root = `folder`

And for `file.java`

- submissionFile = `file.java`
- root  = `file.java`

However, I think the sentences here in the documentation are not very understandable! :D

Reply:
I agree that this is what the comment means, but the code does not seem to do that. Instead, the `root` attribute always points to the directory where the submission (and possibly others) are located, which was given in the CLI. This kind of root cannot be a proper file, the CLI will not allow it.
That's the notion of a root of the CLI.

Other times, like in the `Submission` class, the `submissionRootFile` actually refers to the top-level `File` of the individual submission, which may be a directory or a proper file.
That's the notion of a root of a Submission.

We might want to clear that up a bit.

Reply:
Oh, so this is not the root file of the submission, but rather the root directory specified in the CLI? This is indeed highly confusing!

Reply:
I tried to improve it in https://github.com/jplag/JPlag/pull/2350/commits/63148bbda38d136226f8e6b89fdc4323d5eb1a32, do you think the updated version is clear enough?

Reply:
I would think so, thanks!